--- leveldb/build.gradle	2018-04-22 15:54:45.602122622 +0200
+++ leveldb-patched/build.gradle	2018-04-22 15:53:08.129976729 +0200
@@ -28,6 +28,15 @@
 
   configurations.all {
 }
-  
-  
+
+jar {
+    from {
+        configurations.compile.collect {
+            it.isDirectory() ? it : zipTree(it)
+        }
+        configurations.runtime.collect {
+            it.isDirectory() ? it : zipTree(it)
+        }
+    }
+}
 }
